File: BuySellStock.ts
function BuySellStock(prices: number[]): number {
    let largestDiff:number = 0
    let minSoFar:number = Number.MAX_VALUE
    for (let i = 0; i < prices.length; i++){
        let min = prices[i]
        if (min < minSoFar){
            minSoFar = min
        } else {
            largestDiff = Math.max(largestDiff, min - minSoFar)
        }
    }
    return largestDiff
  }


  console.log(BuySellStock([7, 5, 1, 4, 3, 6, 2]))

File: BuySellStock.py
class Solution:
    def maxProfit(self, prices: list[int]) -> int:
        largestDiff:int = 0
        minSoFar:int = float("inf")
        for min in prices:
            if min < minSoFar:
                minSoFar = min
            else:
                largestDiff = max(largestDiff, min - minSoFar)
        
        return largestDiff


a = Solution()
print(a.maxProfit([7, 5, 2, 4, 3, 6, 2, 8]))